compiler-core/src/error.rs

```diff
@@ -3699,6 +3729,20 @@ fn hint_alternative_operator(op: &BinOp, given: &Type) -> Option<String> {
     }
 }
 
+fn hint_wrap_value_into_result(expected: &Arc<Type>, given: &Arc<Type>) -> Option<String> {
+    let expected = collapse_links(expected.clone());
+    let (expected_ok_type, expected_error_type) = expected.result_types()?;
+
+    let given = collapse_links(given.clone());
+    if collapse_links(expected_ok_type) == given {
```

Comment:
This won't work some times due to the inner types also being links. Good enough for some simple cases though.

Maybe can drop a TODO here? Or implement a `equal` method which is like unify without the mutation? Would remove the need to collapse links too.

Reply:
Done! I've also replace another piece of code that was using `collapse_links` to check for equality of two types